modules/core/src/v2/coins/xlm.ts

```diff
@@ -1,23 +1,148 @@
 import * as _ from 'lodash';
-import { BigNumber } from 'bignumber.js';
+import * as bitcoin from 'bitgo-utxo-lib';
 import * as querystring from 'querystring';
 import * as url from 'url';
 import * as Bluebird from 'bluebird';
-import * as bitcoin from 'bitgo-utxo-lib';
-const co = Bluebird.coroutine;
 import * as request from 'superagent';
 import * as stellar from 'stellar-sdk';
+import { BigNumber } from 'bignumber.js';
 
-import { BaseCoin } from '../baseCoin';
 import { Ed25519KeyDeriver } from '../keyDeriver';
 import * as config from '../../config';
 import * as common from '../../common';
+import {
+  InvalidAddressError,
+  InvalidMemoIdError,
+  KeyRecoveryServiceError,
+  UnexpectedAddressError
+} from '../../errors';
+import { BaseCoin } from '../baseCoin';
+import { NodeCallback } from '../types';
+import { Wallet } from '../wallet';
 
-const maxMemoId = '0xFFFFFFFFFFFFFFFF'; // max unsigned 64-bit number = 18446744073709551615
+const co = Bluebird.coroutine;
+
+interface KeyPair {
+  pub?: string;
+  prv: string;
+}
+
+interface AddressDetails {
+  address: string;
+  memoId?: string;
+}
+
+interface Memo {
+  type: stellar.MemoType;
+  value: string;
+}
+
+interface VerifyAddressOptions {
+  address: string;
+  rootAddress: string;
+}
+
+interface InitiateRecoveryOptions {
+  userKey: string;
+  backupKey: string;
+  recoveryDestination: string;
+  krsProvider?: string;
+  walletPassphrase?: string;
+}
+
+interface RecoveryOptions extends InitiateRecoveryOptions {
+  rootAddress: string;
+}
+
+interface ExplainTransactionOptions {
+  txBase64: string;
+}
+
+interface RecoveryTransaction {
+  tx: string;
+  recoveryAmount: number;
+  backupKey?: string;
+  coin?: string;
+}
+
+interface TransactionPrebuild {
+  txBase64: string;
+}
+
+interface SignTransactionOptions {
+  txPrebuild: TransactionPrebuild;
+  prv: string;
+}
+
+interface HalfSignedTransaction {
+  halfSigned: {
+    txBase64: string;
+  }
+}
+
+interface SupplementGenerateWalletOptions {
+  rootPrivateKey?: string;
+}
+
+interface TransactionOutput {
+  amount: string;
+  address: string;
+}
+
+interface TransactionMemo {
+  value?: string;
+  type?: string;
+}
+
+interface TransactionFee {
+  fee: string;
+  feeRate: number;
+  size: number;
+}
+
+interface TransactionExplanation {
+  displayOrder: string[];
+  id: string;
+  outputs: TransactionOutput[];
+  outputAmount: string;
+  changeOutputs: [];
```

Comment:
this list doesn't have a type

Reply:
There won't ever be any elements of this array, so the type is actually a correct empty array literal. Here's more info if you are curious, since I haven't seen this sort of feature in other languages: https://basarat.gitbooks.io/typescript/docs/types/literal-types.html

Reply:
ok, thanks for clarifying